### ISSUE TITLE
Android tablet MT8163vb chip

### ISSUE THREAD

**Issue by ssbz** [bkerler/mtkclient#234]
This is what I get with seccfg and e metadata,userdata,md_udc. Any ideas on how to add a 2 byte buffer? I think I have a log file or two, I'll dig them up and post them.
C:\mtkclient-main>py mtk da seccfg unlock --preloader=C:\mtkclient-main\preloader_tb8163p3_64_bsp_cu-ip10p.bin
MTK Flash/Exploit Client V1.54 (c) B.Kerler 2018-2021

Preloader - Status: Waiting for PreLoader VCOM, please connect mobile

Port - Hint:

Power off the phone before connecting.
For brom mode, press and hold vol up, vol dwn, or all hw buttons and connect usb.
For preloader mode, don't press any hw button and connect usb.

Preloader
Preloader - [LIB]: ←[31mStatus: Handshake failed, retrying...←[0m
Preloader
Preloader - [LIB]: ←[31mStatus: Handshake failed, retrying...←[0m
Preloader
Preloader - [LIB]: ←[31mStatus: Handshake failed, retrying...←[0m
Port - Device detected :)
Preloader -     CPU:                    MT8163()
Preloader -     HW version:             0x0
Preloader -     WDT:                    0x10007000
Preloader -     Uart:                   0x11002000
Preloader -     Brom payload addr:      0x100a00
Preloader -     DA payload addr:        0x201000
Preloader -     CQ_DMA addr:            0x10212c00
Preloader -     Var1:                   0xb1
Preloader - Disabling Watchdog...
Preloader - HW code:                    0x8163
Preloader - Target config:              0x0
Preloader -     SBC enabled:            False
Preloader -     SLA enabled:            False
Preloader -     DAA enabled:            False
Preloader -     SWJTAG enabled:         False
Preloader -     EPP_PARAM at 0x600 after EMMC_BOOT/SDMMC_BOOT:  False
Preloader -     Root cert required:     False
Preloader -     Mem read auth:          False
Preloader -     Mem write auth:         False
Preloader -     Cmd 0xC8 blocked:       False
Preloader - Get Target info
Preloader - BROM mode detected.
Preloader -     HW subcode:             0x8a00
Preloader -     HW Ver:                 0xcb00
Preloader -     SW Ver:                 0x1
Preloader - ME_ID:                      E43795EB1C939E731E7EC07E84329038
DA_handler - Device is unprotected.
PLTools - Loading payload from mt8163_payload.bin, 0x258 bytes
PLTools - Kamakiri / DA Run
Kamakiri - Trying kamakiri2..
Kamakiri - Done sending payload...
PLTools - Successfully sent payload: C:\mtkclient-main\mtkclient\payloads\mt8163_payload.bin
Port - Device detected :)
DALegacy - Uploading da...
DALegacy - Uploading stage 1...
Preloader - Jumping to 0x200000
Preloader - Jumping to 0x200000: ok.
DALegacy - Got loader sync !
DALegacy - Reading nand info
DALegacy - Reading emmc info
DALegacy - Setting stage 2 config ...
DALegacy - DRAM config needed for : 04010149534f434f4d1129eb20c11719
DALegacy - Reading dram nand info ...
DALegacy - Sending dram info ...
Traceback (most recent call last):
  File "mtk", line 695, in <module>
    mtk = Main(args).run()
  File "C:\mtkclient-main\mtkclient\Library\mtk_main.py", line 510, in run
    mtk = da_handler.configure_da(mtk, preloader)
  File "C:\mtkclient-main\mtkclient\Library\mtk_da_cmd.py", line 102, in configure_da
    if not mtk.daloader.upload_da(preloader=preloader):
  File "C:\mtkclient-main\mtkclient\Library\mtk_daloader.py", line 192, in upload_da
    return self.da.upload_da()
  File "C:\mtkclient-main\mtkclient\Library\mtk_dalegacy.py", line 949, in upload_da
    if self.upload():
  File "C:\mtkclient-main\mtkclient\Library\mtk_dalegacy.py", line 924, in upload
    self.set_stage2_config(self.config.hwcode)
  File "C:\mtkclient-main\mtkclient\Library\mtk_dalegacy.py", line 821, in set_stage2_config
    checksum = unpack(">H", self.usbread(2))[0]  # 0x440C
struct.error: unpack requires a buffer of 2 bytes

C:\mtkclient-main>

**Comment by bkerler** [comment]
That means that sending the dram config fails, much likely the preloader dram config doesn't match your device.

**Comment by ssbz** [comment]
I had come to the same conclusion. Wrong preloader. I have looked at a
couple preloaders that were in 3 firmwares I downloaded for it in notepad,
and all were for different devices of the same brand. I noticed in the
target configuration for CDC communication and data at the last it is
either in Chinese or states error reading string data. When I first used
your app it would blue screen my system when I plugged in my tablet. I
switched driver's around, took out usbdk and used zadig. Tried several mtk
vcom driver's until it got to where it functioned as it does now. The
tablet is hard bricked, and a tech from hovatek I hired for an hour
downloaded your files to unbrick it. It was still bricked at the end, but I
continued trying myself. I have an associate's degree in digital
electronics, and know just enough programming to brick a tablet good. I'm
sure it is my stupidity and not your app that's the issue here. It is a
cheap 60 dollar onn tablet and I have spent about 40 trying to unbrick it.
If your app gets it done I will give up 20 to support your project. It is
worth the learning experience. I appreciate any help I can get here. The
tablet is an Onn 8 surf, model tbbvnc100005207. I can't seem to get the
right firmware, although I do have an ota update downloaded that seems
right. Thanks again.
